File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    
    lowestPrice := prices[0]
	maxProfit := 0
	
	for _,p := range prices {
		if p < lowestPrice {
			lowestPrice = p
		} else if (p - lowestPrice) > maxProfit {
			maxProfit = p - lowestPrice
		}
	}

	return maxProfit
    
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int leastPrice = prices[0];
        int maxProfit = 0;
        for(int currPrice : prices) {
            if(currPrice <= leastPrice) {
                leastPrice = currPrice;
            } else {
                int currProfit = currPrice - leastPrice;
                if (currProfit > maxProfit) {
                    maxProfit = currProfit;
                }
            }
        }

        return maxProfit;
    }
}